sktime/forecasting/trend/_spline_trend_forecaster.py

```diff
@@ -0,0 +1,215 @@
+        if not self.prediction_intervals:
+            raise ValueError(
+                "Prediction intervals were not calculated during fit. \
+                Set prediction_intervals=True at initialization."
+            )
+
+        # 1. get X (design matrix) and M = (X^t X)^-1
+        t_train = _get_X_numpy_int_from_pandas(self.train_index_).flatten()
+        X = np.polynomial.polynomial.polyvander(t_train, self.degree)
+        if not self.with_intercept:
+            X = X[:, 1:]  # remove the column of 1's that handles the intercept
+
+        M = np.linalg.inv(X.T @ X)
+
+        # 2. get time vector t for the forecast horizons
+        if fh.is_relative:
+            fh = fh.to_absolute(cutoff=self.train_index_[-1])
+
+        t_fh = fh.to_pandas()
+        fh_periods = _get_X_numpy_int_from_pandas(t_fh)
+        t = np.array(fh_periods)
+
+        # 3. calculate (half-) range of PI (1 + sqrt(x_0^t M x_0)) (up to scaling)
+        start = 0 if self.with_intercept else 1
+        v = []
+
+        for _, z in enumerate(t):
+            w = np.array([z**j for j in range(start, self.degree + 1)])
+            v.append(w.T @ M @ w)
+
+        v = (1 + np.array(v)).flatten()  # see Hyndman FPP3 Section 7.9
+
+        l_var = v * self.s_squared_  # see Hyndman FPP3 Section 7.9
+        pred_var = pd.DataFrame(l_var, columns=[self._y.name])
+        return pred_var
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return ``"default"`` set.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            ``MyClass(**params)`` or ``MyClass(**params[i])`` creates a valid test
+            instance.
+            ``create_test_instance`` uses the first (or only) dictionary in ``params``
+        """
+        from sklearn.ensemble import RandomForestRegressor
+
+        params_list = [
+            {},
+            {
+                "regressor": RandomForestRegressor(),
+                "degree": 2,
+                "with_intercept": False,
+                "prediction_intervals": False,
+            },
+            {
+                "regressor": RandomForestRegressor(),
+                "degree": 2,
+                "with_intercept": True,
+                "prediction_intervals": True,
+            },
+            {
+                "regressor": RandomForestRegressor(),
+                "degree": 2,
+                "with_intercept": False,
+                "prediction_intervals": True,
+            },
+        ]
```

Comment:
Without increasing the number of test paramas, can you cover more init arguments like extrapolation or knots please?